src/main/java/net/dv8tion/jda/internal/entities/AbstractChannelImpl.java

```diff
@@ -62,4 +64,69 @@ public T setName(String name)
         this.name = name;
         return (T) this;
     }
+
+    // -- Union Hooks --
+
+    @Nonnull
+    public PrivateChannel asPrivateChannel()
+    {
+        return transformChannel(PrivateChannel.class);
+    }
+
+    @Nonnull
+    public TextChannel asTextChannel()
+    {
+        return transformChannel(TextChannel.class);
+    }
+
+    @Nonnull
+    public NewsChannel asNewsChannel()
+    {
+        return transformChannel(NewsChannel.class);
+    }
+
+    @Nonnull
+    public VoiceChannel asVoiceChannel()
+    {
+        return transformChannel(VoiceChannel.class);
+    }
+
+    @Nonnull
+    public StageChannel asStageChannel()
+    {
+        return transformChannel(StageChannel.class);
+    }
+
+    @Nonnull
+    public ThreadChannel asThreadChannel()
+    {
+        return transformChannel(ThreadChannel.class);
+    }
+
+    @Nonnull
+    public GuildMessageChannel asGuildMessageChannel()
+    {
+        return transformChannel(GuildMessageChannel.class);
+    }
+
+    @Nonnull
+    public StandardGuildChannel asStandardGuildChannel()
+    {
+        return transformChannel(StandardGuildChannel.class);
+    }
+
+    @Nonnull
+    public StandardGuildMessageChannel asStandardGuildMessageChannel()
+    {
+        return transformChannel(StandardGuildMessageChannel.class);
+    }
+
+    private <TOut extends Channel> TOut transformChannel(Class<TOut> toObjectClass)
+    {
+        if (toObjectClass.isInstance(this))
+            return toObjectClass.cast(this);
+
+        String cleanedClassName = this.getClass().getSimpleName().replace("Impl", "");
```

Comment:
Use `this.getType()` instead?

Reply:
This gives a "TextChannel" or "VoiceChannel".
`getType()` would give TEXT or VOICE, so this was an intentional choice given how it is used for user messaging.

Reply:
The problem is that this locks us into the concrete types we have now, if we ever make some kind of more specialized type, we would end up breaking this name. Isn't the channel type clear enough?

Reply:
We've had the same naming scheme for our Interface -> Impl classes since 2015. I'm not overly concerned that we're gonna get this wrong.

I don't think the channel type is clear enough, especially since threads are collapsed into ThreadChannel whereas their type is actually 3 different types.

Generally, the point of the unions is to get people comfortable knowing what interfaces can be casted to what other interfaces, so I want these errors to convey that same thing.